python_modules/libraries/dagster-k8s/dagster_k8s/launcher.py

```diff
@@ -445,8 +445,10 @@ def check_run_worker_health(self, run: DagsterRun):
                 WorkerStatus.FAILED, "Run has not completed but K8s job has no active pods"
             )
 
-        if status.failed:
-            return CheckRunHealthResult(WorkerStatus.FAILED, "K8s job failed")
         if status.succeeded:
             return CheckRunHealthResult(WorkerStatus.SUCCESS)
+        else:
+            if status.failed > 0 and not status.active:
```

Comment:
what's your confidence level that there's no time period between when the first pod fails and the second pod starts  up due to the backoff where this would still fire? Is the transition 'atomic' in that way?

Reply:
from k8s API specs, sounds like it'll only be 0 for finished jobs. the second pod may have a startup delay, but it should be pending in that time and therefore count towards this: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.32/#jobstatus-v1-batch.

This happens regularly in our cluster, so my confidence will increase dramatically once I see this change running there. I don't think it's unsafe for other users in the meantime, but happy to try a beta version if we think that's the right direction.